File: app/src/main/java/ru/ra66it/updaterforspotify/domain/interactors/SpotifyInteractor.java
package ru.ra66it.updaterforspotify.domain.interactors;

import javax.inject.Inject;

import io.reactivex.Observable;
import ru.ra66it.updaterforspotify.domain.models.Spotify;
import ru.ra66it.updaterforspotify.domain.repositories.SpotifyRepository;

public class SpotifyInteractor {

    private SpotifyRepository spotifyRepository;

    @Inject
    public SpotifyInteractor(SpotifyRepository spotifyRepository) {
        this.spotifyRepository = spotifyRepository;
    }

    public Observable<Spotify> getLatestSpotify() {
        return spotifyRepository.getLatestSpotify();
    }
}

File: app/src/main/java/ru/ra66it/updaterforspotify/domain/interactors/SpotifyInteractor.kt
package ru.ra66it.updaterforspotify.domain.interactors

import ru.ra66it.updaterforspotify.data.repositories.SpotifyRepository
import ru.ra66it.updaterforspotify.domain.model.Result
import ru.ra66it.updaterforspotify.domain.model.Spotify
import javax.inject.Inject

class SpotifyInteractor @Inject constructor(
        private val spotifyRepository: SpotifyRepository
) {

    suspend fun getSpotify(): Result<Spotify> {
        return spotifyRepository.getSpotify()
    }
}
